plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -271,4 +288,99 @@ public static HoodieTableFileSystemView getFileSystemView(
         return new HoodieTableFileSystemView(
                 tableMetadata, metaClient, metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
     }
+
+    public static HiveHudiPartitionInfo buildHiveHudiPartitionInfo(HudiTableHandle tableHandle, String partitionName, Partition partition)
+    {
+        return new HiveHudiPartitionInfo(
+                tableHandle.getSchemaTableName(),
+                Location.of(tableHandle.getBasePath()),
+                partitionName,
+                partition,
+                tableHandle.getPartitionColumns(),
+                tableHandle.getPartitionPredicates());
+    }
+
+    public static Partition buildPartition(String partitionPath, HudiTableHandle tableHandle, PartitionValueExtractor partitionValueExtractor)
+    {
+        if (partitionPath == null || partitionPath.isEmpty()) {
+            return Partition.builder()
+                    .setDatabaseName(tableHandle.getSchemaName())
+                    .setTableName(tableHandle.getTableName())
+                    .withStorage(storageBuilder ->
+                            storageBuilder.setLocation(tableHandle.getBasePath())
+                                    .setStorageFormat(StorageFormat.NULL_STORAGE_FORMAT))
+                    .setColumns(ImmutableList.of())
+                    .setValues(ImmutableList.of())
+                    .build();
+        }
+        else {
+            List<String> values = partitionValueExtractor.extractPartitionValuesInPath(partitionPath);
+            List<HiveColumnHandle> hiveColumnHandles = tableHandle.getPartitionColumns();
+
+            if (hiveColumnHandles.size() != values.size()) {
+                throw new TrinoException(HUDI_INVALID_PARTITION_VALUE, "Invalid partition path: " + partitionPath);
+            }
+
+            List<Column> columns = hiveColumnHandles.stream()
+                    .map(column -> new Column(
+                            column.getName(),
+                            column.getHiveType(),
+                            column.getComment(),
+                            Collections.emptyMap()))
+                    .toList();
```

Comment:
The partition columns are the same across different partitions, so we could build the partition list once per table handle instead of per partition?